src/utils/Utils.cpp

```diff
@@ -39,11 +41,16 @@ QString Utils::sanitizeForHttpSeparators(const QString& input)
 /////////////////////////////////////////////////////////////////////////////////////////
 QString Utils::ComputerName()
 {
+  static std::once_flag flag;
+  static QString name;
+  std::call_once(flag, [](){
 #ifdef Q_OS_MAC
-  return OSXUtils::ComputerName();
+    name = OSXUtils::ComputerName();
 #else
-  return QHostInfo::localHostName();
+    name = QHostInfo::localHostName();
 #endif
+  });
+  return name;
 }
```

Comment:
Is there any kind of scenario where the computer name changes and we don't want to require a PMP restart? I guess not?